net/BUILD.gn

```diff
@@ -1175,6 +1175,16 @@ component("net") {
       "socket/tcp_socket_starboard.h",
       "socket/udp_socket_starboard.cc",
       "socket/udp_socket_starboard.h",
+      # Use POSIX socket for SB16
```

Comment:
Why not filter these lists using `sb_api_version`, then you don't need to modify the source code with `#if SB_API_VERSION >= 16` and friends?

Reply:
There is a chance it will be merged back to Cobalt25, and it looks more consistent with other source files when checking on SB_API versions.

Reply:
If you have seen that the same usage exists where entire files are only used for certain starboard versions that could be filtered in the gn config, then please file a bug that we can clean those and this instance here up as a followup. Minimizing changes in code we take from upstream makes rebasing easier.